src/main/java/com/coveo/spillway/storage/RedisStorage.java

```diff
@@ -65,96 +62,9 @@
   private final JedisPool jedisPool;
   private final String keyPrefix;
 
-  public RedisStorage(URI uri) {
-    this(uri, DEFAULT_PREFIX);
-  }
-
-  public RedisStorage(URI uri, String prefix) {
-    this(uri, prefix, Protocol.DEFAULT_TIMEOUT);
-  }
-
-  public RedisStorage(URI uri, String prefix, int timeout) {
-    this(new GenericObjectPoolConfig(), uri, prefix, timeout);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, URI uri) {
-    this(poolConfig, uri, DEFAULT_PREFIX);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, URI uri, String prefix) {
-    this(poolConfig, uri, prefix, Protocol.DEFAULT_TIMEOUT);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, URI uri, String prefix, int timeout) {
-    jedisPool = new JedisPool(poolConfig, uri, timeout);
-    keyPrefix = prefix;
-  }
-
-  public RedisStorage(String host) {
-    this(host, Protocol.DEFAULT_PORT);
-  }
-
-  public RedisStorage(String host, int port) {
-    this(host, port, DEFAULT_PREFIX);
-  }
-
-  public RedisStorage(String host, int port, String prefix) {
-    this(new GenericObjectPoolConfig(), host, port, prefix);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, String host) {
-    this(poolConfig, host, Protocol.DEFAULT_PORT);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, String host, int port) {
-    this(poolConfig, host, port, DEFAULT_PREFIX);
-  }
-
-  public RedisStorage(GenericObjectPoolConfig poolConfig, String host, int port, String prefix) {
-    this(poolConfig, host, port, prefix, Protocol.DEFAULT_TIMEOUT);
-  }
-
-  public RedisStorage(
-      GenericObjectPoolConfig poolConfig, String host, int port, String prefix, int timeout) {
-    jedisPool = new JedisPool(poolConfig, host, port, timeout);
-    keyPrefix = prefix;
-  }
-
-  public RedisStorage(
-      GenericObjectPoolConfig poolConfig,
-      String host,
-      int port,
-      String prefix,
-      int timeout,
-      String password) {
-    this(poolConfig, host, port, prefix, timeout, password, Protocol.DEFAULT_DATABASE);
-  }
-
-  public RedisStorage(
-      GenericObjectPoolConfig poolConfig,
-      String host,
-      int port,
-      String prefix,
-      int timeout,
-      String password,
-      int database) {
-    jedisPool = new JedisPool(poolConfig, host, port, timeout, password, database);
-    keyPrefix = prefix;
-  }
-
-  @Deprecated
-  public RedisStorage(Jedis jedis) {
-    this(jedis, DEFAULT_PREFIX);
-  }
-
-  @Deprecated
-  public RedisStorage(Jedis jedis, String prefix) {
-    this(
-        new GenericObjectPoolConfig(),
-        jedis.getClient().getHost(),
-        jedis.getClient().getPort(),
-        prefix,
-        jedis.getClient().getConnectionTimeout());
+  public RedisStorage(Builder builder) {
```

Comment:
Hmm or maybe the constructor for RedisStorage should take all the needed stuff as arguments? But it'd still need to be package level if we want to preserve API compatibility when adding params.